processing/p5.js
# Added .gitattributes to Increase compatability with Window users and line endings

Resolves #6207 

 Changes:
Added a file, .gitattributes, which forces line endings to LF with default Windows settings which would typically change them to CRLF. I've tested this on my own device and everything worked smoothly(Screenshot below) It also worked on @snwarner22's device with one error due to a test taking too long to run on his device.

I saw that another Issue and PR were closed 6 years ago, #2484 , which also could have fixed the problem. I think it’s really a compatibility issue. This `.gitattributes` file would make the library more compatible with windows and overall easier for new contributors to use. I'd like to ask that the community reconsiders adding this file so future users don’t encounter this issue and potentially get deterred from working on p5.js. 

The windows portion of my team encountered this and found another fix: `git config --global core.autocrlf false`, but this would impact the user's whole device and might be difficult for others to discover. I'd also like to note I've run this 3 times on my device all with all tests passing.

I'd also like to mention that I'm working from the main branch as that was the best way to ensure the file was working properly. I understand it's best to work from a feature branch and hope this doesn't cause any issues.

# TESTING!
Testing has been requested and would be very helpful. If anyone on Windows is open to testing all you need to do is:

1.  Run the following  : `git config --global core.autocrlf true`, in your terminal to ensure default settings are in place. 
2. Run `git clone https://github.com/SilasVM/p5.js.git`
3. Enter your new clone using `cd p5.js`
4. In your new clone run `npm ci`
5. Next run `npm test` and let us know the results in the comments

 # Screenshots of the change:
![gitattributes](https://github.com/processing/p5.js/assets/124199231/909b48c5-118f-420f-95f3-ef1f4a7051c2)

Screenshot of `npm test` running successfully with default windows settings: `core.autocrlf true`:
![Success](https://github.com/processing/p5.js/assets/124199231/49b41060-7c5d-483d-972f-7ecb164253a8)

@snwarner22's  Screenshot of 'npm test' running with the exception of one timeout error:
![skwarner22](https://github.com/processing/p5.js/assets/124199231/16ae0838-a335-4115-afc9-f8d947df329b)

Article where info was pulled from: 

https://dev.to/deadlybyte/please-add-gitattributes-to-your-git-repository-1jld#gitattributes-reset

File Authors

Co-authored-by: Karl R. Wurst <karl@w-sts.com>
Co-authored-by: Stoney Jackson <dr.stoney@gmail.com>
Co-authored-by: Silas Morgan <victorsmorgan@gmail.com>

Reviewers:
Sekani Warner <wsekani@gmail.com>

#### PR Checklist


- [X] [Inline documentation] is included / updated
- [X] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests
